plugins/registry.js

```diff
@@ -701,4 +705,39 @@ export default class PluginRegistry {
 
         return id;
     }
+
+    // INTERNAL: Subject to change without notice.
+    // DANGER: Interferes with historic routes.
+    // Register a global header menu item
+    // Accepts the following:
+    // - baseURL - The route to be displayed at starting from the siteURL
+    // - switcherIcon - A react element to display as the icon in the product switcher
+    // - switcherText - A string or React element to display in the product switcher
+    // - switcherLinkURL - A string specifying the URL the switcher item should point to.
+    // - mainComponent - The component to be displayed below the global header when your route is active.
+    // - headerComponent - A component to fill the generic area in the center of
+    //                     the global header when your route is active.
+    // All parameters are required.
+    // Returns a unique identifier.
+    registerProduct(baseURL, switcherIcon, switcherText, switcherLinkURL, mainComponent, headerComponent) {
+        const id = generateId();
+
+        store.dispatch({
+            type: ActionTypes.RECEIVED_PLUGIN_COMPONENT,
+            name: 'Product',
+            data: {
+                id,
+                pluginId: this.id,
+                switcherIcon: resolveReactElement(switcherIcon),
+                switcherText: resolveReactElement(switcherText),
+                baseURL: '/' + standardizeRoute(baseURL),
```

Comment:
I thought it more clear to keep that external. You are right we just add it everywhere, but it doesn't look as nice above `this.id + fixedRoute` rather than `this.id + '/' + fixedRoute`.